src/descriptor/create_descriptor.rs

```diff
@@ -1,391 +0,0 @@
-/// to a `Stack` and outputs a `Wsh` descriptor. Does not check miniscript inside the descriptor
-/// 5. `Bare`: translates script_sig to a `Stack` and script_pubkey to miniscript
-/// 6. `Sh`: Checks redeem_script hash, translates remaining elements from script_sig to
-/// a `Stack` and redeem script to miniscript. Does not check the miniscript
-/// 7. `ShWpkh`: Checks redeem_script hash, translates remaining elements from script_sig to
-/// a `Stack` and validates `Wpkh` sig, pubkey.
-/// 8. `ShWsh`: Checks witness script hash, pops witness script and converts it to miniscript.
-/// translates the remaining witness to a `Stack`
-pub fn from_txin_with_witness_stack<'txin>(
-    script_pubkey: &bitcoin::Script,
-    script_sig: &'txin bitcoin::Script,
-    witness: &'txin [Vec<u8>],
-) -> Result<(Descriptor<bitcoin::PublicKey>, Stack<'txin>), Error> {
-    if script_pubkey.is_p2pk() {
-        verify_p2pk(script_pubkey, script_sig, witness)
-    } else if script_pubkey.is_p2pkh() {
-        verify_p2pkh(script_pubkey, script_sig, witness)
-    } else if script_pubkey.is_v0_p2wpkh() {
-        let (pk, stack) = verify_p2wpkh(script_pubkey, script_sig, witness)?;
-        Ok((Descriptor::Wpkh(pk), stack))
-    } else if script_pubkey.is_v0_p2wsh() {
-        let (ms, stack) = verify_wsh(script_pubkey, script_sig, witness)?;
-        Ok((Descriptor::Wsh(ms), stack))
-    } else if script_pubkey.is_p2sh() {
-        let (redeem_script, stack) = verify_p2sh(script_pubkey, script_sig)?;
-        if redeem_script.is_v0_p2wpkh() {
-            //Therefore while calling verify_wpkh, an argument of Script::new() is passed instead
-            //of script_sig. The redeem_script becomes the script_pubkey
-            let (pk, stack) = verify_p2wpkh(&redeem_script, &Script::new(), witness)?;
-            Ok((Descriptor::ShWpkh(pk), stack))
-        } else if redeem_script.is_v0_p2wsh() {
-            //Therefore while calling verify_wpkh, an argument of Script::new() is passed instead
-            //of script_sig. The redeem_script becomes the script_pubkey
-            let (ms, stack) = verify_wsh(&redeem_script, &Script::new(), witness)?;
-            Ok((Descriptor::ShWsh(ms), stack))
-        } else {
-            if !witness.is_empty() {
-                return Err(Error::NonEmptyWitness);
-            }
-            let ms = Miniscript::<bitcoin::PublicKey, Legacy>::parse(&redeem_script)?;
-            Ok((Descriptor::Sh(ms), stack))
-        }
-    } else {
-        //bare
-        let stack: Result<Vec<stack::Element>, Error> = script_sig
-            .instructions_minimal()
-            .map(stack::Element::from_instruction)
-            .collect();
-        if !witness.is_empty() {
-            return Err(Error::NonEmptyWitness);
-        }
-        let ms = Miniscript::<bitcoin::PublicKey, Bare>::parse(script_pubkey)?;
-        Ok((Descriptor::Bare(ms), Stack::from(stack?)))
-    }
-}
-
-#[cfg(test)]
```

Comment:
We are losing these tests. I don't know if it is worth fixing these in this PR, but these can be adapted to the new `from_txdata` API in the future.

Reply:
Added a commit with replacement for these tests